src/txn_builder.rs

```diff
@@ -992,7 +992,7 @@ impl<L: Ledger> TransactionState<L> {
         spec: TransferSpec<'k>,
         proving_keys: &'k KeySet<TransferProvingKey<'a>, key_set::OrderByOutputs>,
         rng: &mut ChaChaRng,
-    ) -> Result<TransferInfo<L>, TransactionError> {
+    ) -> Result<(TransferNote, TransactionInfo<L>), TransactionError> {
```

Comment:
Looks like `TransactionInfo` is never returned alone without `TransferNote`? If that's the case, can we add a `note: TranserNote` filed to the `TransactionInfo` struct?